ethereum/sidecar/attestation_validation.go

```diff
@@ -0,0 +1,156 @@
+package sidecar
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	ethconfig "github.com/keep-network/keep-common/pkg/chain/ethereum"
+	ethconnect "github.com/mezo-org/mezod/ethereum"
+	"github.com/mezo-org/mezod/ethereum/bindings/portal"
+)
+
+var (
+	ErrInvalidAttestation      = errors.New("invalid attestation")
+	ErrValidatorNotInTheBitmap = errors.New("validatorId not in the bitmap yet")
+)
+
+type AttestationValidation struct {
+	bridgeContract ethconnect.BridgeContract
+	address        common.Address
+}
+
+func NewAttestationValidation(
+	bridgeContract ethconnect.BridgeContract,
+	validatorAddress common.Address,
+) *AttestationValidation {
+	return &AttestationValidation{
+		bridgeContract: bridgeContract,
+		address:        validatorAddress,
+	}
+}
+
+func (av *AttestationValidation) IsConfirmed(
+	attestation *portal.MezoBridgeAssetsUnlocked,
+) error {
+	ok, err := av.bridgeContract.ValidateAssetsUnlocked(*attestation)
+	if err != nil {
+		return fmt.Errorf("couldn't call validateAssetsUnlocked: %w", err)
+	}
+	if !ok {
+		// this is a specific case of error
+		// the attestation is not valid, we specify it down the line
+		return ErrInvalidAttestation
+	}
+
+	ok, err = av.bridgeContract.ConfirmedUnlocks(attestation.Amount)
+	if err != nil {
+		return fmt.Errorf("couldn't get confirmedLocks: %w", err)
+	}
+	if ok {
+		return nil
+	}
+
+	return av.checkOwnConfirmation(attestation)
+}
+
+func (av *AttestationValidation) WaitForConfirmation(
+	blockHeightWaiter ethconfig.BlockHeightWaiter,
+	startBlock, confirmations uint64,
+	attestation *portal.MezoBridgeAssetsUnlocked,
+) error {
+	return WaitForBlockConfirmations(
+		blockHeightWaiter,
+		startBlock,
+		confirmations,
+		func() error {
+			return av.checkOwnConfirmation(attestation)
+		},
+	)
+}
+
+func (av *AttestationValidation) checkOwnConfirmation(
+	attestation *portal.MezoBridgeAssetsUnlocked,
+) error {
```

Comment:
Let's also make it `(true, error)` to clearly distinct the result state and errors.